lib/sinatra/base.rb

```diff
@@ -1834,7 +1834,7 @@ def self.force_encoding(data, *) data end
 
     reset!
 
-    set :environment, (ENV['RACK_ENV'] || :development).to_sym
+    set :environment, (ENV['SINATRA_ENV'] || :development).to_sym
```

Comment:
I agree with the idea of this change but perhaps there's a middle ground as this would currently be a significant breaking change as it stands:

```ruby
set :environment, (ENV['SINATRA_ENV'] || ENV['RACK_ENV'] || :development).to_sym
```

Then the use of `RACK_ENV` could be deprecated and checking it removed in the next major version.